client/me/purchases/remove-purchase/index.jsx

```diff
@@ -154,34 +159,42 @@ const RemovePurchase = React.createClass( {
 				.catch( err => debug( err ) ); // shouldn't get here
 		}
 
-		this.props.removePurchase( purchase.id, user.get().ID ).then( () => {
-			const productName = getName( purchase );
-
-			if ( isDomainRegistration( purchase ) ) {
-				notices.success(
-					this.translate( 'The domain {{domain/}} was removed from your account.', {
-						components: { domain: <em>{ productName }</em> }
-					} ),
-					{ persistent: true }
-				);
-			} else {
-				notices.success(
-					this.translate( '%(productName)s was removed from {{siteName/}}.', {
-						args: { productName },
-						components: { siteName: <em>{ selectedSite.domain }</em> }
-					} ),
-					{ persistent: true }
-				);
-			}
+		this.props.removePurchase( purchase.id, user.get().ID )
+			.then( () => {
+				const productName = getName( purchase );
 
-			page( purchasePaths.purchasesRoot() );
-		} ).catch( () => {
-			this.setState( { isRemoving: false } );
+				if ( isDomainRegistration( purchase ) ) {
+					if ( isDomainOnlySite ) {
+						// removing the domain from a domain-only site deletes the site entirely
+						receiveDeletedSite( selectedSite );
```

Comment:
I wonder if we shouldn't specify in the comment that the site is deleted by the server. This would clarify why we call `receiveDeletedSite()` instead of `deleteSite()`.